controller/registry/dockerclient.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""Support the Deis workflow by manipulating and publishing Docker images."""
+
+from __future__ import unicode_literals
+import io
+import logging
+
+from django.conf import settings
+from rest_framework.exceptions import PermissionDenied
+from simpleflock import SimpleFlock
+import docker
+
+logger = logging.getLogger(__name__)
+
+
+class DockerClient(object):
+    """Use the Docker API to pull, tag, build, and push images to deis-registry."""
+
+    FLOCKFILE = '/tmp/controller-pull'
+
+    def __init__(self):
+        self.client = docker.Client(version='auto')
+        self.registry = settings.REGISTRY_HOST + ':' + str(settings.REGISTRY_PORT)
```

Comment:
would it make sense to re-purpose `settings.REGISTRY_URL` here? We can probably omit the `http://` from the URL anyways since Docker doesn't explicitly care about the protocol.